woocommerce/woocommerce
# Feature/fix build

So this came out of a hack week project that I've started. It is basically a progression to make our build process and local development more developer friendly.

**Features:**
* All min files are now properly ignored so that you won't get any untracked files warning when working locally. This is evident when using `grunt watch` and other scripts.
* All JS files are now included in the Uglify process of Grunt so that it is consistent with other JS files we currently Uglify. This also prevents an accidental change in the main JS file that may have forgotten to minify.
* Add woocommerce.zip to gitignore as this comes up as untracked file when running the build script locally.
* Added `npm run test:unit` to run our unit tests.

**Removed:**
* Removed all the JS min files as these will now all be built with the build/watch script.

**Testing**
* Remove all min files and run `npm run build-watch` and ensure all min files are created.
* Remove all min files and run `npm run build` and ensure all min files are included and in the zip.
* Ensure we don't have any missing min files for each vendor packages we're using.
* Ensure you're not seeing any untracked files for the JS and zip files.
